### PR TITLE
Fix Issue 22454 - OpenBSD: Add prototypes for pthread_np.h

### PR DESCRIPTION
**Pull request by ibara**

### CODE REVIEW

**Comment by ibara** [comment]
Spurious `dlang-community/dfmt` test error?

File: src/core/sys/openbsd/pthread_np.d [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  */
+module core.sys.openbsd.pthread_np;
+
+version (OpenBSD):
+extern (C):
+nothrow:
+@nogc:
+
+public import core.sys.posix.sys.types;
+public import core.sys.posix.signal; // for stack_t
```
Comment (kinke) [inline src/core/sys/openbsd/pthread_np.d:16]: The other BSD variants don't publicly import this module; I guess non-publicly importing `stack_t` alone would be better wrt. consistent namespace pollution.

Comment (ibara) [reply src/core/sys/openbsd/pthread_np.d]: I do plan to use this module in one place in druntime after this is committed.
I switched `public import core.sys.posix.signal; // for stack_t` to `private import core.sys.posix.signal : stack_t;` in a force push.

Comment (kinke) [reply src/core/sys/openbsd/pthread_np.d]: Thx - `private` is the default and so not needed; there are only half a dozen other occurrences of `private import` in druntime/Phobos, probably ancient.

Comment (ibara) [reply src/core/sys/openbsd/pthread_np.d]: Alright; I'll remove the private keyword. Thanks.

Comment (ibara) [reply src/core/sys/openbsd/pthread_np.d]: (Done)

**Review by kinke — APPROVED** [review]

**Comment by kinke** [comment]
[Wrt. dfmt, that's totally unrelated, see https://github.com/dlang-community/dfmt/pull/544.]

**Pull request merged**